lib/util/package-rules/match.ts

```diff
@@ -0,0 +1,30 @@
+import is from '@sindresorhus/is';
+import { minimatch } from 'minimatch';
+import { logger } from '../../logger';
+import { regEx } from '../regex';
+
+export function matchRegexOrMinimatch(pattern: string, input: string): boolean {
+  if (pattern.length > 2 && pattern.startsWith('/') && pattern.endsWith('/')) {
+    try {
+      const regex = regEx(pattern.slice(1, -1));
+      return regex.test(input);
+    } catch (err) {
+      logger.once.warn({ err, pattern }, 'Invalid regex pattern');
+      return false;
+    }
+  }
+  return minimatch(input, pattern, { dot: true });
```

Comment:
Note: no massage from `*` to `.*`. We should deprecate that support and instead migrate it in config. I think `**` works for minimatch (matches both (`abc` and `abc/def`);

Reply:
Also I can't see any point matching `*` repositories here anyway

Reply:
🤔 This functiom causes the re-validate the pattern for each input again and again. that's bad for overall perf. we should  have a global cache for them. It's not only we iterate oer arrays, we also evaluate the package rules multiple times on a single repo run

Reply:
The regEx constructor is cached though. Which bit do you have concerns on?

Reply:
the minimatch parsing, as it's regex internally too, so we should cache minimatch instances like we do for regex.

Reply:
Does minimatch support such caching? I'm not sure this level of optimization is necessary for this PR especially as it will be a rarely used match anyway

Reply:
no buildin cache, we could wrap it like we do for regex.

we do use minimatch a lot, so I think it can benefit from a cache.
but right, this can be done later.